### ISSUE TITLE
Feature: Lists should be toggleable

### ISSUE THREAD

**Issue by BennyAlex** [facebook/lexical#6733]
## Description

If you dispatch a list command like INSERT_UNORDERED_LIST_COMMAND it will not toggle a list back to normal.
This is different from formatting a text with bold, italic etc.

So I want to recreate the keyboard shortcuts from word. Using Ctrl + Shift + L can toggle a list. 
This should be the default behavior as users a used to it.

It could also be solved by adding a seperate toggle command.

I think text alignment it is the same.
Also it is the same for headlines (the dropdown in the playground).
If you select the same style again, it should be reverted.

**Comment by etrepum** [comment]
You could certainly do this on your own by registering the command at a higher priority, detecting if the selection is on a list, and then removing the list. Since lists can be nested I think either approach is probably valid, and it would be harder to override the toggle logic if that was the default.

**Comment by BennyAlex** [comment]
https://support.microsoft.com/en-us/office/keyboard-shortcuts-in-word-95ef89dd-7142-4b50-afb2-f762f663ceb2#bkmk_frequentwin

Sure here is an uncomplete list:

Decrease the font size by 1 point. | Ctrl+Left bracket ([)
Increase the font size by 1 point. | Ctrl+Right bracket (])
Center the text. | Ctrl+E
Align the text to the left. | Ctrl+L
Align the text to the right. | Ctrl+R
Indent the paragraph. | Ctrl+M
Remove a paragraph indent. | Ctrl+Shift+M
Center the paragraph. | Ctrl+E
Justify the paragraph. | Ctrl+J
Align the paragraph to the left. | Ctrl+L
Align the paragraph to the right. | Ctrl+R
Indent the paragraph. | Ctrl+M
Remove a paragraph indent. | Ctrl+Shift+M
Create a hanging indent. | Ctrl+T
Remove a hanging indent. | Ctrl+Shift+T
Remove paragraph formatting. | Ctrl+Q
Apply the Normal style. | Ctrl+Shift+N
Apply the Heading 1 style. | Ctrl+Alt+1
Apply the Heading 2 style. | Ctrl+Alt+2
Apply the Heading 3 style. | Ctrl+Alt+3
Apply subscript formatting. | Ctrl+Equal sign ( = )
Apply superscript formatting. | Ctrl+Shift+Plus sign (+)

**Comment by BennyAlex** [comment]
One suggestion was just adding a toggle command so its there out of the box.

I cant create a nested list inside the playground and also not in word using the keyboard shortcut so I think having the same behaviour would be the best.

**Comment by etrepum** [comment]
Looks like you're right that the command does not allow you to create nested lists, but the implementation does allow it and that's how multiple levels work. For example if you look at [this document](https://playground.lexical.dev/#doc=H4sIAAAAAAAAE82U3WrDMAyFX6Wca1PiMmjrpxh0d2UXbqx1BsUethIWSt59OClsHWOUkkKvbB2hIz7_6ARyXmLaiRWCOSHFKGWt3z27RAFm_1_gSKxnmErhLabGyrhtoiMYhKIwFLL0XAQoCH0KDA4tM8mCfZYi9h8lPeYUOkrZxwCjh1cF5xPVMsZgSfhuVOx8cBSmpmeTYumFmgsjhc5yS8VS3Z0ttM2BErn56PQj0VEWcov5IVdXQ974Kn7ZFellyk4wI65N5_O2RxhExvy3tHpUgGs_0R0ApnnwB0B7G8A4xi7e4aDANsvOduRg9Hq1rTZbXT2ttVbIsU11qXtm2x9TbIP7WY1qqTfLCsMX7uKcqS8FAAA) you can see several list nodes nested in listitem nodes

**Comment by BennyAlex** [comment]
Yes I dont say the are wrong or should not be possbile but using a keyboard short cut or the command should toggle the current list I guess

**Comment by kirandash** [comment]
I've investigated this issue and can confirm that list toggling is working as expected. The keyboard shortcut `Cmd + Option + 5` (or `Ctrl + Alt + 5` on Windows/Linux) correctly toggles numbered lists. This is implemented in the codebase and verified by test cases.

The functionality is properly implemented in:
1. The keyboard shortcuts are defined in `ShortcutsPlugin/shortcuts.ts`
2. The list toggling is handled in `ShortcutsPlugin/index.tsx`
3. There are test cases in `List.spec.mjs` that verify the toggling behavior

The issue appears to be resolved as the current implementation does support toggling lists back to their original state.

Please let me know if I am missing anything?

**Comment by etrepum** [comment]
Seems fixed to me as well 👍
